protocol-tests/src/main/java/fixtures/headexceptions/implementation/HeadExceptionsImpl.java

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package fixtures.headexceptions.implementation;
+
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.Head;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import fixtures.headexceptions.models.CustomizedException;
+import reactor.core.publisher.Mono;
+
+/** An instance of this class provides access to all the operations defined in HeadExceptions. */
+public final class HeadExceptionsImpl {
+    /** The proxy service used to perform REST calls. */
+    private final HeadExceptionsService service;
+
+    /** The service client containing this operation class. */
+    private final AutoRestHeadExceptionTestServiceClientImpl client;
+
+    /**
+     * Initializes an instance of HeadExceptionsImpl.
+     *
+     * @param client the instance of the service client containing this operation class.
+     */
+    HeadExceptionsImpl(AutoRestHeadExceptionTestServiceClientImpl client) {
+        this.service =
+                RestProxy.create(HeadExceptionsService.class, client.getHttpPipeline(), client.getSerializerAdapter());
+        this.client = client;
+    }
+
+    /**
+     * The interface defining all the services for AutoRestHeadExceptionTestServiceHeadExceptions to be used by the
+     * proxy service to perform REST calls.
+     */
+    @Host("{$host}")
+    @ServiceInterface(name = "AutoRestHeadExceptio")
+    public interface HeadExceptionsService {
+        @Head("/http/success/200")
+        @ExpectedResponses({200})
+        @UnexpectedResponseExceptionType(
+                value = ResourceNotFoundException.class,
+                code = {404})
+        @UnexpectedResponseExceptionType(CustomizedException.class)
+        Mono<Response<Void>> head200(@HostParam("$host") String host, RequestOptions requestOptions, Context context);
```

Comment:
`CustomizedException` is the class that supposed to be manually create by dev, as grow-up.